core/engine/src/context/time.rs

```diff
@@ -179,6 +187,68 @@ impl Clock for FixedClock {
     }
 }
 
+/// The `DynamicTimeZoneProvider` is a wrapper type that allows users
```

Comment:
Why not use `Box<dyn TimeZoneProvider>` instead? We don't need a wrapper in this case, since the provider is only accessible within the crate